src/shared/logs-show.c

```diff
@@ -1278,7 +1259,38 @@ int show_journal_entry(
         if (r < 0)
                 return r;
 
-        r = output_funcs[mode](f, j, mode, n_columns, flags, fields, highlight);
+        JOURNAL_FOREACH_DATA_RETVAL(j, data, length, r) {
+                r = parse_fieldv(data, length, message_fields, ELEMENTSOF(message_fields));
+                if (r < 0)
+                        return r;
+        }
+        if (r == -EBADMSG) {
+                log_debug_errno(r, "Skipping message we can't read: %m");
+                return 0;
+        }
+        if (r < 0)
+                return log_error_errno(r, "Failed to get journal fields: %m");
+
+        if (realtime)
+                r = safe_atou64(realtime, &ts.realtime);
+        if (!realtime || r < 0 || !VALID_REALTIME(ts.realtime))
+                r = sd_journal_get_realtime_usec(j, &ts.realtime);
+        if (r < 0)
+                ts.realtime = 0;  /* Set realtime to zero which is interpreted as invalid */
+
+        if (monotonic)
+                r = safe_atou64(monotonic, &ts.monotonic);
+        if (!monotonic || r < 0 || !VALID_MONOTONIC(ts.monotonic))
+                r = sd_journal_get_monotonic_usec(j, &ts.monotonic, &boot_id);
+        if (r < 0)
+                ts.monotonic = 0;  /* Set monotonic to zero which is interpreted as invalid */
+
+        /* Restart all data before handover to output function */
+        sd_journal_restart_data(j);
+        sd_journal_restart_unique(j);
+        sd_journal_restart_fields(j);
```

Comment:
mayb split this all out in a new helper `journal_get_dual_timestamp()` or so.

Reply:
Moved into `get_dual_timestamp()`